tcms/kiwi_auth/views.py

```diff
@@ -82,38 +81,41 @@ def get(self, request):
         return render(request, self.template_name, {'form': self.form_class()})
 
 
-@require_GET
-def confirm(request, activation_key):
+class Confirm(RedirectView):
     """Confirm the user registration"""
 
-    # Get the object
-    try:
-        _activation_key = UserActivationKey.objects.select_related('user')
-        _activation_key = _activation_key.get(activation_key=activation_key)
-    except UserActivationKey.DoesNotExist:
+    http_method_names = ['get']
+
+    def get_redirect_url(self, *args, **kwargs):
+        self.url = self.request.GET.get('next', reverse('core-views-index'))
+        activation_key = kwargs['activation_key']
+        try:
+            _activation_key = UserActivationKey.objects.select_related('user')
+            _activation_key = _activation_key.get(activation_key=activation_key)
+        except UserActivationKey.DoesNotExist:
+            messages.add_message(
+                self.request,
+                messages.ERROR,
+                _('This activation key no longer exists in the database')
+            )
+            return super().get_redirect_url(*args, **kwargs)
+
+        if _activation_key.key_expires <= timezone.now():
+            messages.add_message(self.request, messages.ERROR, _('This activation key has expired'))
+            return super().get_redirect_url(*args, **kwargs)
+
+        # All thing done, start to active the user and use the user login
+        user = _activation_key.user
```

Comment:
you may also use `_activation_key.user.is_active` directly instead of using an intermediate variable. It doesn't make the code more readable and we don't use the `user` variable anywhere else.